examples/Consumer/Program.cs

```diff
@@ -231,15 +152,38 @@ public static void Run_ManualAssign(string brokerList, List<string> topics)
                 consumer.OnError += (_, error)
                     => Console.WriteLine($"Error: {error}");
 
-                // Raised on deserialization errors or when a consumed message has an error != NoError.
-                consumer.OnConsumeError += (_, error)
-                    => Console.WriteLine($"Consume error: {error}");
-
                 while (true)
                 {
-                    if (consumer.Consume(out Message<Ignore, string> msg, TimeSpan.FromSeconds(1)))
+                    try
+                    {
+                        var cr = consumer.Consume();
+                        if (cr.Message != null)
+                        {
+                            Console.WriteLine($"Received message at {cr.TopicPartitionOffset}: ${cr.Message}");
+                        }
+
+                        // Alternative:
+                        //
+                        // switch (cr.Error.Code)
```

Comment:
well pointed out. the factoring of this, and related things isn't quite right yet. a refactor is coming. i expect there will only be exceptions here for error cases.

Reply:
to answer your question, information is only passed in the error field over the exception if the error is not error like. In the current iteration, I've removed the error field from `ConsumeResult` (it's only present on the corresponding exception type). Same on the produce side, though there, we need a delivery report type both with and without the error because error notification differs between `ProduceAsync` and `BeginProduce`.